pyo3-build-config/src/impl_.rs

```diff
@@ -638,6 +627,20 @@ pub fn cross_compiling(
         // No cross-compiling environment variables set; try to determine if this is a known case
         // which is not cross-compilation.
 
+        if host.starts_with(&target_triple) {
+            // Not cross-compiling if arch-vendor-os is all the same
+            // e.g. x86_64-unknown-linux-musl on x86_64-unknown-linux-gnu host
+            //      x86_64-pc-windows-gnu on x86_64-pc-windows-msvc host
+            return Ok(None);
+        }
+
+        dbg!(&target_triple, &host);
+
+        if target_triple == "i686-pc-windows" && host.starts_with("x86_64-pc-windows") {
```

Comment:
Minor nit: `target_tiple` seems to be `format!`ed unconditionally but is only used within the if clause